incremental-font-transfer/src/font_patch.rs

```diff
@@ -12,20 +12,25 @@
 
 use crate::patchmap::{PatchEncoding, PatchUri};
 
-use read_fonts::tables::ift::{CompatibilityId, TablePatchFlags};
-use read_fonts::tables::ift::{TableKeyedPatch, TablePatch};
+use crate::glyph_keyed::apply_glyph_keyed_patch;
+use read_fonts::tables::ift::{
+    CompatibilityId, GlyphKeyedPatch, TableKeyedPatch, TablePatch, TablePatchFlags,
+};
 use read_fonts::types::Tag;
 use read_fonts::{FontData, FontRef};
 use read_fonts::{FontRead, ReadError};
 use shared_brotli_patch_decoder::{shared_brotli_decode, DecodeError};
 use std::collections::BTreeSet;
+
 use write_fonts::FontBuilder;
 
 /// An incremental font patch which can be used to extend a font.
 ///
 /// See: <https://w3c.github.io/IFT/Overview.html#font-patch-formats>
+#[derive(Clone)]
 pub enum IncrementalFontPatch<'a> {
     TableKeyed(TableKeyedPatch<'a>),
+    GlyphKeyed(GlyphKeyedPatch<'a>),
```

Comment:
looking at the code below, instead of using `&[IncrementalFontPatch]` (with the condition that there can be only one table-keyed patch but multiple glyph-keyed patches) do you think it would make more sense to try and represent that in the type system here, somehow?

What I mean is ending up with something like,

```rust
pub enum IncrementalFontPatch<'a, 'b> {
     TableKeyed(TableKeyedPatch<'a>),
     GlyphKeyed(&'b [GlyphKeyedPatch<'a>]),
}
```

which is annoying for the extra lifetime but might otherwise make life easier? But maybe not, it's not totally clear to me at what stage you would have the slice.

Reply:
Interesting idea, but IncrementalFontPatch's are derived from a single PatchUri + binary data via a .into_patch() method on the URI. So we'd need to modify the API for deriving the patches from uri's and I think it would end up kind of clunky, and ultimately I think this just shifts the compatibility check up one level. Either way the caller of the API is ultimately responsible for not feeding in an incompatible set of patches.

Reply:
I think maybe a better approach is to have two distinct apply_patch methods that take either a single TableKeyed or multiple GlyphKeyed and leave it up to the caller to extract the appropriate types from the generic IncrementalFontPatch list (like I mentioned it's already something a caller will need to do before calling this API).

Reply:
I'll give that a try and see what it ends up looking like.

Reply:
Looking into this more I think we probably want more broad changes to the entire set of IFT APIs, so I'm going to leave this as is for now but revisit it later with a more intentional design for the full API surface.